pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/elastic/export/generate/MappingGenerator.java

```diff
@@ -65,10 +66,13 @@ public Map<String, Object> getCompleteMap(Object propertyMap) {
     }
 
     @Override
-    protected Map<String, Object> parseJson() {
+    protected Map<String, Object> parseJson() throws DsmNullDateException {
         Map<String, Object> resultMap = new HashMap<>();
         Map<String, Object> fieldsByValues = parseJsonToMapFromValue();
-
+        if (fieldsByValues == null) {
+            //This happens when the value is null, and here dsm throws an exception so that the null value is processed as single values
+            throw new DsmNullDateException("fieldsByValues is null");
```

Comment:
See this [comment](https://github.com/broadinstitute/ddp-study-server/pull/2884/files#r1628545574)

Reply:
Why do we need a separate exception class here?  Based on a null `fieldsByValue`, how do we know that what's missing was a date vs. something else?  The fact this exception is caught and handled in the exact same way as `JsonParseException` in BaseGenerator makes me think that the separate exception class does not lead to different exception handling logic.

Reply:
Excellent point. Here’s how `fieldsByValue` is calculated: we either return a Map or throw one of two different exceptions.

The only way the Map could be null is if the value itself was null. Until today, we handled null values by converting them to blank strings in the frontend. However, now that we need to handle null values specifically for dates, I thought this should be addressed by a separate exception.

What we actually want is the same outcome as when the value couldn't be parsed into a map. Here’s the relevant code:

```protected Map<String, Object> parseJsonToMapFromValue() {
        try {
            return ObjectMapperSingleton.instance().readValue(String.valueOf(getNameValue().getValue()), Map.class);
        } catch (com.fasterxml.jackson.core.JsonParseException | JsonMappingException je) {
            throw new JsonParseException(je.getMessage());
        } catch (IOException e) {
            throw new RuntimeException(e);
        }
    }
```

I think one way to maybe help with your concerns could be to only throw this exception if the  `getNameValue().getName()` is one of the date fields for fax sents, because those are the only fields we could be sending nulls for